docs/testing/e2e_test_suite.rst

```diff
@@ -0,0 +1,249 @@
+Tests are executed in real browsers using the W3C WebDriver protocol, with Selenium managing browser interactions.
+
+Here’s an overview of tests directory structure:
+
+.. image:: images/e2e_test_suite.png
+    :width: 300
+    :alt: Tests directory structure
+
+.. list-table::
+   :header-rows: 1
+
+   * - Directory
+     - Description
+   * - ``_data/``
+     - Contains fixture data used in tests, including SQL dump files and sample CSV files.
+   * - ``_output/``
+     - Contains output from tests in case of failures. This includes snapshots of the browser in JPEG format and generated HTML reports for troubleshooting.
+   * - ``_support/``
+     - 
+       - ``AcceptanceTester.php``: Contains login logic that runs before each test.
+       - ``Helper/``: Stores custom helper functions. For example, DbHelper.php automates the process of generating SQL dump files and populating the database. It prepares the database from scratch if no dump file exists and exports a SQL file for future use.
+       - ``Page/``: Stores UI locators for each page. Avoid hard-coding complex CSS or XPath locators in tests; instead, use PageObject classes.
+       - ``Step/``: Contains step objects that group common functionalities for tests.
+   * - ``acceptance/``
+     - Contains acceptance tests.
+
+.. vale off
+
+Writing and Running Tests
+**************************
+
+.. vale on
+
+Writing tests
+=============
+
+Writing tests in Codeception involves creating files within the tests/Acceptance directory. Each file contains a class with methods that define the test scenarios.
+
+1. Create a New Test File: Use the following command to generate a new file:
+
+.. code-block:: bash
+
+    bin/codecept generate:cest acceptance <TestName>
+
+This command will create a file named TestSuiteNameCest.php inside the tests/Acceptance directory.
+
+2. Define Test Scenarios: Open the generated file and define your test scenarios. Each method within the class represents a different scenario. Use Codeception's built-in assertions and helper functions to verify the expected outcomes. Here’s a simple example:
+
+.. code-block:: PHP
+
+    <?php
+
+    class TestSuiteNameCest
+    {
+        public function _before(AcceptanceTester $I)
+        {
+            // Code to run before each test
+        }
+
+        public function _after(AcceptanceTester $I)
+        {
+            // Code to run after each test
+        }
+
+        // Define your test methods
+
+        public function login(AcceptanceTester $I)
+        {
+            $I->amOnPage('/s/login');
+            $I->fillField('#username', $name);
+            $I->fillField('#password', $password);
+            $I->click('button[type=submit]');
+            $I->see('Dashboard');
+        }
+    }
+
+3. Utilize PageObjects and StepObjects:
+
+Organize your tests by using PageObject and StepObject classes. This keeps your tests clean and maintainable by separating locators and test steps into reusable components.
```

Comment:
Please be consistent with your use of colons and line breaks - elsewhere the text is on the same line but here it's a new line. Let's pick one or the other and stick to it, I'd recommend previewing it and deciding what works best from a layout / readability perspective.

Reply:
I think a new line would be better for readability. What do you think?

Reply:
Could you share a screenshot of the rendered output with/without?

Reply:
Option 1:
(with new line)
<img width="400" alt="image" src="https://github.com/user-attachments/assets/1b195fd9-a83d-4ff4-a0cd-4b969ed70964">

Option 2:
(with same line)
<img width="400" alt="image" src="https://github.com/user-attachments/assets/8405ddfb-3a51-4464-b2fb-e4d9a3200842">

Reply:
I agree, I think it's more readable on a new line. If we do that I don't think we need the colon at the end of the title/heading though. Otherwise, all good!